src/main/java/org/candlepin/resource/ProductResource.java

```diff
@@ -101,43 +112,57 @@ private Product fetchProduct(String productUuid) {
         return product;
     }
 
-    @Override
-    public Stream<ProductDTO> getProducts(Integer page, Integer perPage, String order, String sortBy) {
-        PageRequest pageRequest = ResteasyContext.getContextData(PageRequest.class);
-
-        ProductQueryArguments queryArgs = new ProductQueryArguments();
-
-        long count = this.productCurator.getProductCount(queryArgs);
-
-        if (pageRequest != null) {
-            Page<Stream<ProductDTO>> pageResponse = new Page<>();
-            pageResponse.setPageRequest(pageRequest);
-
-            if (pageRequest.isPaging()) {
-                queryArgs.setOffset((pageRequest.getPage() - 1) * pageRequest.getPerPage())
-                    .setLimit(pageRequest.getPerPage());
-            }
-
-            if (pageRequest.getSortBy() != null) {
-                boolean reverse = pageRequest.getOrder() == PageRequest.DEFAULT_ORDER;
-                queryArgs.addOrder(pageRequest.getSortBy(), reverse);
-            }
-            pageResponse.setMaxRecords((int) count);
-
-            // Store the page for the LinkHeaderResponseFilter
-            ResteasyContext.pushContext(Page.class, pageResponse);
-        }
-        // If no paging was specified, force a limit on amount of results
-        else {
-            int maxSize = config.getInt(ConfigProperties.PAGING_MAX_PAGE_SIZE);
-            if (count > maxSize) {
-                String errmsg = this.i18n.tr("This endpoint does not support returning more than {0} " +
-                    "results at a time, please use paging.", maxSize);
-                throw new BadRequestException(errmsg);
-            }
+    /**
+     * Retrieves an Owner instance for the owner with the specified key/account. If a matching owner could
+     * not be found, this method throws an exception.
+     *
+     * @param key
+     *  The key for the owner to retrieve
+     *
+     * @throws NotFoundException
+     *  if an owner could not be found for the specified key.
+     *
+     * @return
+     *  the Owner instance for the owner with the specified key.
+     *
+     * @httpcode 200
+     * @httpcode 404
+     */
+    private Owner resolveOwner(String key) {
+        Owner owner = this.ownerCurator.getByKey(key);
+        if (owner == null) {
+            throw new NotFoundException(i18n.tr("Owner with key \"{0}\" was not found.", key));
         }
 
-        return this.productCurator.listAll(queryArgs).stream()
+        return owner;
+    }
+
+    @Override
+    @Transactional
+    // GET /products
```

Comment:
Double checking to see if this comment was intentionally left here

Reply:
Yes. I was planning to start doing that so I didn't have to jump between two files to figure out what endpoint I was looking at.

Reply:
Yeah, that makes sense! I'll start doing that too because I'm always having to jump between api spec file and the resource files as well to make sure I'm looking at the right resource method for the endpoint